### PR TITLE
chore(commonjs): switch to fdir for fewer dependencies

### PR DESCRIPTION
**Pull request by benmccann**
## Rollup Plugin Name: `@rollup/plugin-commonjs`

This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [x] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

### Description

Note that this incorporates https://github.com/rollup/plugins/pull/1740

`glob` pulls in 27 dependencies: https://npmgraph.js.org/?q=glob
`fdir` pulls in none: https://npmgraph.js.org/?q=fdir

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,5 +86,5 @@
       "test/"
     ]
   },
-  "packageManager": "pnpm@8.15.8+sha512.d1a029e1a447ad90bc96cd58b0fad486d2993d531856396f7babf2d83eb1823bb83c5a3d0fc18f675b2d10321d49eb161fece36fe8134aa5823ecd215feed392"
```
Comment (kytta) [inline package.json:89]: **issue:** Even though this might seem like an improvement, this change is out of scope of this PR and also causes a complete re-rendering of pnpm-lock.yaml (due to lockfile version change)

**suggestion:** Remove this line and re-install packages using pnpm v8.15.8

File: packages/commonjs/package.json
Comment (kytta) [inline packages/commonjs/package.json:1]: **suggestion:** `fdir` requires `picomatch` if one wants to use `fdir().glob(...)`. I assume it exists elsewhere in the dependency tree, but let's be explicit and add it here, too

**Review by kytta — CHANGES_REQUESTED** [review]
Hey there! Please note that **I am not a Rollup contributor**, and I don't have write access to this repository. Nevertheless, I am interested in this PR being merged, so I voluntarily reviewed it.

As stated in the comments, I can imagine the sudden PNPM version switch to be a problem for the maintainers. As such, I advise you to use an older PNPM version, so that this PR focuses explicitly on the glob -> fdir transition 😄

**Comment by kytta** [comment]
Okay, I'm embarrassed that I haven't read the PR description properly 🤦🏻‍♂️ I didn't realise this includes a different PR which would update PNPM to v9 ... Ignore that part of my review, then :D

File: packages/commonjs/package.json
Comment (benmccann) [reply packages/commonjs/package.json:1]: done
